server/main.go

```diff
@@ -62,20 +65,21 @@ func main() {
 		log.Fatalf("Failed to create meter: %v", err)
 	}
 
-	// If the user has set a database user or password, init the database
-	if os.Getenv("DB_USER") != "" || os.Getenv("DB_PASSWORD") != "" {
-		config.InitDB()
-	}
+	var proxyPort int
+	var resourceRepository interfaces.IRepository
```

Comment:
You can create a function that takes the InMemoryDb flag as a parameter and returns the interface. Define which database client (pg/inmemory) is used in that function

Reply:
If I understand you correctly, the idea would be to create a function like

```
resourceRepository := getRepository(*InMemoryDB Flag){
    if(*InMemoryDB){
        return  rsRepo.NewMemoryRepository()
    }
    return rsRepo.NewRepository(config.DB)
} 
```
Because of the logic on lines 75 - 79, I actually think it may be preferrable to keep as is. 

This server was originally known as the Three-In-One-Server because it was mashed together. It needs another general refactor and I'll suggest we make the "In Memory" vs "Database" logic flow improved after we ratchet to the next set point (i.e., after we get your stats into production)

Reply:
yeah we can refactor it later...